### PR TITLE
[FLINK-14468][docs] Update Kubernetes docs

### PR DESCRIPTION
**Pull request by mbode**
- use Deployment from apps/v1 instead of extensions/v1beta1
- run as unprivileged user

## What is the purpose of the change

*This pull request updates the Kubernetes documentation to use the current `Deployment` resource and run unprivileged*

## Brief change log

- *Docs update*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):  no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`:  no
  - The serializers:  no 
  - The runtime per-record code paths (performance sensitive):  no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no 
  - The S3 file system connector:  no 

## Documentation

  - Does this pull request introduce a new feature?  no

### CODE REVIEW

File: docs/ops/deployment/kubernetes.md [hunk 1]
```diff
@@ -230,6 +240,8 @@ spec:
         volumeMounts:
         - name: flink-config-volume
           mountPath: /opt/flink/conf/
+        securityContext:
+          runAsUser: 9999
```
Comment (wangyang0918) [inline docs/ops/deployment/kubernetes.md]: Why securityContext need to be set? What will happen if the uid(9999) does not exist.

Comment (mbode) [reply docs/ops/deployment/kubernetes.md]: I stumbled over this trying to deploy the example templates to a Kubernetes cluster where I did not have permissions to run privileged workloads. It seems to me to be a best practice to run unprivileged if possible.
The user _flink_ with uid _9999_ is present in the official Flink docker image, e.g. [here](https://github.com/docker-flink/docker-flink/blob/2e4b45b10e8efe04c324e44cacf7df16b2553f0f/1.9/scala_2.12-debian/Dockerfile#L55) for Flink 1.9/Scala 2.12.

Comment (aljoscha) [reply docs/ops/deployment/kubernetes.md]: Maybe we should then add a comment that this refers to a user ID that exists in the Flink docker images and that users should change it if necessary.

Comment (wangyang0918) [reply docs/ops/deployment/kubernetes.md]: Agree. We should add a comment here.

Comment (aljoscha) [reply docs/ops/deployment/kubernetes.md]: @mbode any update here?

Comment (mbode) [reply docs/ops/deployment/kubernetes.md]: Whoops, must have missed it, sorry! Just pushed the comment like you proposed.

Comment (aljoscha) [reply docs/ops/deployment/kubernetes.md]: Cool, thanks! ☕️

**Comment by aljoscha** [comment]
Thanks for your work!

**Pull request merged**